### DISCUSSION TITLE
how to wait for and grab chat message

### DISCUSSION THREAD

**Discussion by kingultron99** [diamondburned/arikawa#315] [Help] [answered]
Either this is not possible, or so easy that its staring me in the face and im severely overthinking this.

I'm trying to implement a system where the user presses a component button, then the bot waits for a chat message from the same user in the same channel, takes that input and processes it.

I have considered the use of `api.messages` and using the latest message it returns, which works, however the problem is that its not actually waiting for a message to be sent, its just grabbing a pre-existing message and spitting it out (and only runs once).

Are there any half decent ways of implementing a system like this?

**Comment by diamondburned** [comment, marked as answer]
Add a `*MessageCreateEvent` handler as usual. This is demonstrated in the `simple` example.

If you want to wait for a specific message from the user, you could use `ChanFor` and add a new handler on demand, or you can use a single handler and an internal registry of waiting users.
